homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -330,9 +332,33 @@ async def _async_replace_device(self, replace_device):
             if new_entity_id is not None:
                 entity_migration_map[new_entity_id] = entry
 
+        @callback
+        def _handle_state_change(entity_id: str, old_state, new_state):
+            # Wait for entities to finish cleanup
+            if new_state is None and entity_id in pending_entities:
+                pending_entities.remove(entity_id)
+            if not pending_entities:
+                wait_for_entities.set()
+
+        # Create a set with entities to be removed which are currently in the state
+        # machine
+        pending_entities = {
+            entry.entity_id
+            for entry in entity_migration_map.values()
+            if not self.hass.states.async_available(entry.entity_id)
+        }
+        wait_for_entities = asyncio.Event()
+        track_state_changes = async_track_state_change(
```

Comment:
I'd clarify the name so it's easier to understand that the `track_state_changes` function unsubscribes the state change listener. Right now, it sounds like it starts tracking changes.